### PR TITLE
[doc] Update website docs link

### LINKED ISSUE #10160: [Bug] [doc] Inconsistent links between Chinese and English documents on the website
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.

### What happened

1. Inconsistent links between Chinese and English documents on the website
2. The most recent release is V2.0.5 not V1.3.9

### What you expected to happen

1. Chinese and English document links are consistent
2. Update the latest version to 2.0.5

### How to reproduce

ds-min

### Anything else

_No response_

### Version

3.0.0-alpha

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by SnowMoon-Dev**
## Brief change log

close [#10160](https://github.com/apache/dolphinscheduler/issues/10160)

## Verify this pull request
Manually verified the change by testing locally

### CODE REVIEW

**Comment by zhongjiajie** [comment]
I am looking on it

**Comment by zhongjiajie** [comment]
ref: apache/dolphinscheduler-website#788, and #10124. We should remove `dev` and `pydolphinscheudler` under the `DOCS`  and  I just forget it in #10124. and about which version we should put in the second after version `3.0.0-beta-1` we need further discuss.  I prefer still use version `1.3.9` until we fix the bug in `2.0.5` release `2.0.6` which is more stable. WDYT @lenboo @caishunfeng @songjianet @nielifeng @dailidong

File: docs/configs/site.js [hunk 1]
```diff
@@ -50,14 +50,24 @@ export default {
...
+          {
+            key: 'pythonAPI',
+            text: 'PyDolphinScheduler',
+            link: '/python/index.html',
+          },
+          {
+            key: 'docsdev',
+            text: 'dev',
+            link: '/en-us/docs/dev/user_doc/about/introduction.html',
+          },
```
Comment (zhongjiajie) [inline docs/configs/site.js]:
```suggestion
```

Comment (zhongjiajie) [reply docs/configs/site.js]: we should delete them in Chinese version config

**Comment by caishunfeng** [comment]
>I prefer still use version `1.3.9` until we fix the bug in `2.0.5` release `2.0.6` which is more stable. 

Agree with you.

File: docs/configs/site.js (hunk 1 above)
Comment (SnowMoon-Dev) [reply docs/configs/site.js]: ok， I will delete it in Chinese documentation

**Comment by SnowMoon-Dev** [comment]
> 

ok i agree with you too, after waiting for the 2.0.5 bug to be fixed, put 2.0.6

**Review by zhongjiajie — APPROVED** [review]
LGTM, thanks ~

**Pull request merged**